src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -211,26 +213,32 @@ private boolean areTotalsEqual(InvoiceLine invoiceLine1, InvoiceLine invoiceLine
         validateInvoiceLine(invoice, invoiceLine, invoiceLineFromStorage);
         invoiceLine.setInvoiceLineNumber(invoiceLineFromStorage.getInvoiceLineNumber());
 
-        // Just persist updates if invoice is already finalized
-        if (isPostApproval(invoice)) {
-          return updateInvoiceLineToStorage(invoiceLine);
-        }
+        return protectionHelper.isOperationRestricted(invoice.getAcqUnitIds(), UPDATE)
```

Comment:
Think [this](https://github.com/folio-org/mod-invoice/pull/70/files#r319538531) comment applies here too. We can now use [foreign key cross table index queries](https://github.com/folio-org/raml-module-builder/tree/master#cql2pgjson-foreign-key-cross-table-index-queries) instead of getting an invoiceLine -> invoice -> invoice.getAcqUnitIds

Reply:
An invoice and a invoice line from storage are needed in `validateInvoiceLine` and `applyAdjustmentsAndUpdateLine`.